src/main/java/edu/kit/compiler/codegen/pattern/BinaryInstructionPattern.java

```diff
@@ -16,30 +16,43 @@
 import lombok.RequiredArgsConstructor;
 
 @RequiredArgsConstructor
-public class BinaryInstructionPattern implements Pattern<InstructionMatch> {
+public abstract class BinaryInstructionPattern<T extends Operand.Target, S extends Operand.Source>
+        implements Pattern<InstructionMatch> {
 
     private final ir_opcode opcode;
-    private final String command;
-    private final Pattern<? extends OperandMatch<? extends Operand.Target>> left;
-    private final Pattern<? extends OperandMatch<? extends Operand.Source>> right;
-    private final boolean hasMemory;
-
+    private final Pattern<? extends OperandMatch<T>> targetPattern;
+    private final Pattern<? extends OperandMatch<S>> sourcePattern;
+    private final int offset;
+    private final boolean commutate;
+
+    public static <T extends Operand.Target, S extends Operand.Source> BinaryInstructionPattern<T, S> of(
+            ir_opcode opcode, String command, Pattern<? extends OperandMatch<T>> target,
+            Pattern<? extends OperandMatch<S>> source, int offset, boolean commutate) {
+        return new BinaryInstructionPattern<T, S>(opcode, target, source, offset, commutate) {
+            @Override
+            protected InstructionMatch getMatch(Node node, OperandMatch<T> target,
+                    OperandMatch<S> source, MatcherState matcher) {
+                var targetRegister = getTarget(target.getOperand(),
+                        () -> matcher.getNewRegister(source.getOperand().getSize()));
+                return new BinaryInstructionMatch(node, command, target, source, targetRegister);
```

Comment:
Is there a specific reason why this is an anonymous subclass instead of `command` being a member of the class?

Reply:
This is a remnant of the patterns I tried to implement for Mul and Div replacement. Seems I forgot to remove this change to `BinaryInstruction` with the rest of the changes.